tools/update-docs.js

```diff
@@ -24,46 +24,100 @@ const last = require('lodash/last')
 const rules = require('./lib/rules')
 const categories = require('./lib/categories')
 
+const ROOT = path.resolve(__dirname, '../docs/rules')
+
 function formatItems (items) {
   if (items.length <= 2) {
     return items.join(' and ')
   }
   return `all of ${items.slice(0, -1).join(', ')} and ${last(items)}`
 }
 
-const ROOT = path.resolve(__dirname, '../docs/rules')
-for (const rule of rules) {
-  const filePath = path.join(ROOT, `${rule.name}.md`)
-  const categoryIndex = categories.findIndex(category => category.categoryId === rule.meta.docs.category)
-  const title = `# ${rule.meta.docs.description} (${rule.ruleId})`
-  const notes = []
-
-  if (categoryIndex >= 0) {
-    const presets = categories.slice(categoryIndex).map(category => `\`"plugin:vue/${category.categoryId}"\``)
-    notes.push(`- :gear: This rule is included in ${formatItems(presets)}.`)
+class DocFile {
+  constructor (rule) {
+    this.rule = rule
+    this.filePath = path.join(ROOT, `${rule.name}.md`)
+    this.content = fs.readFileSync(this.filePath, 'utf8')
   }
-  if (rule.meta.deprecated) {
-    if (rule.meta.docs.replacedBy) {
-      const replacedRules = rule.meta.docs.replacedBy.map(name => `[vue/${name}](${name}.md) rule`)
-      notes.push(`- :warning: This rule was **deprecated** and replaced by ${formatItems(replacedRules)}.`)
+
+  static read (rule) {
+    return new DocFile(rule)
+  }
+
+  write () {
+    fs.writeFileSync(this.filePath, this.content)
+  }
+
+  updateHeader () {
+    const { ruleId, meta } = this.rule
+    const categoryIndex = categories.findIndex(category => category.categoryId === meta.docs.category)
+    const title = `# ${meta.docs.description} (${ruleId})`
```

Comment:
What do you think about preparing header to look like this?
![image](https://user-images.githubusercontent.com/4093559/48940927-a36f4780-ef19-11e8-812e-cecfeb1aa2e4.png)

Reply:
i like this

Reply:
i did some digging, and looks like we will have to write/override css to have something like this.